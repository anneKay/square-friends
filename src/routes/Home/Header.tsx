import React, { FC } from "react";
import { Row, Col, Form, FormControl, InputGroup } from "react-bootstrap";
import logo from "../../assets/images/logo.png";
import userIcon from "../../assets/images/user.png";
import messageIcon from "../../assets/images/message.png";
import notificationIcon from "../../assets/images/notification.png";
import profile from "../../assets/images/profile.png";
import dropdown from "../../assets/images/dropdown.png";
import searchIcon from "../../assets/images/search.png";
import funnelIcon from "../../assets/images/funnel.png"
import "../../assets/stylesheet/home/header.scss";

const Header: FC<any> = () => {
  return (
    <>
      <Row className="body-header">
        <Col> <h2 className="header-text">Friends List</h2></Col>
        <Col className="justify-content-end">
          <Row>
            <Col sm={2}><img className="search-icon" src={searchIcon} alt="search"></img></Col>
            <Col sm={6}>
            <InputGroup>
              <FormControl type="text" placeholder="Sort by:" className=" mr-sm-2">
              </FormControl>
              <InputGroup.Prepend>
              <InputGroup.Text><img src={dropdown} alt="dropdown"/></InputGroup.Text>
            </InputGroup.Prepend>
            </InputGroup>
            </Col>
            <Col><img className="filter-icon" src={funnelIcon} alt="filter" /></Col>
          </Row>
          
          
        </Col>
    </Row>
    </>
  )
}

export default Header;
