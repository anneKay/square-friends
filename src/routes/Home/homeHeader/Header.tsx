import React, { FC } from "react";
import { Row, Col, FormControl, InputGroup } from "react-bootstrap";
import { Translate } from "react-localize-redux";
import dropdown from "../../../assets/images/dropdown.png";
import searchIcon from "../../../assets/images/search.png";
import funnelIcon from "../../../assets/images/funnel.png";
import "../../../assets/stylesheet/home/header.scss";


export interface Props {
}

const Header: FC<Props> = () => {
  return (
    <>
      <Row xs={1} md={2}className="body-header">
        <Col> <h2 className="header-text"><Translate id="header.title" >Friends List</Translate></h2></Col>
        <Col className="justify-content-end">
          <Row className="header-input">
            <Col sm={2}><img className="search-icon" src={searchIcon} alt="search" /></Col>
            <Col sm={6}>
            <InputGroup>
              <FormControl type="text" placeholder="Sort by:" className=" mr-sm-2">
              </FormControl>
              <InputGroup.Prepend>
              <InputGroup.Text><img src={dropdown} alt="dropdown"/></InputGroup.Text>
            </InputGroup.Prepend>
            </InputGroup>
            </Col>
            <Col><img className="filter-icon" src={funnelIcon} alt="filter" /></Col>
          </Row>
        </Col>
    </Row>
    </>
  )
}

export default Header;
