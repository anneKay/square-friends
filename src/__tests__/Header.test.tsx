import React from "react";
import { render } from "@testing-library/react";

import Header from "../routes/Home/homeHeader/Header";

describe("<Header />", () => {
  it('renders correctly', () => {
    const tree = render(<Header />)
    expect(tree).toMatchSnapshot();
  });
});
