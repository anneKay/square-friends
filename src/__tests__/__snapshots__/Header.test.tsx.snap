// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="body-header row row-cols-md-2 row-cols-1"
      >
        <div
          class="col"
        >
           
          <h2
            class="header-text"
          >
            Missing translationId: header.title for language: \${ languageCode }
          </h2>
        </div>
        <div
          class="justify-content-end col"
        >
          <div
            class="header-input row"
          >
            <div
              class="col-sm-2"
            >
              <img
                alt="search"
                class="search-icon"
                src="search.png"
              />
            </div>
            <div
              class="col-sm-6"
            >
              <div
                class="input-group"
              >
                <input
                  class=" mr-sm-2 form-control"
                  placeholder="Sort by:"
                  type="text"
                />
                <div
                  class="input-group-prepend"
                >
                  <span
                    class="input-group-text"
                  >
                    <img
                      alt="dropdown"
                      src="dropdown.png"
                    />
                  </span>
                </div>
              </div>
            </div>
            <div
              class="col"
            >
              <img
                alt="filter"
                class="filter-icon"
                src="funnel.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="body-header row row-cols-md-2 row-cols-1"
    >
      <div
        class="col"
      >
         
        <h2
          class="header-text"
        >
          Missing translationId: header.title for language: \${ languageCode }
        </h2>
      </div>
      <div
        class="justify-content-end col"
      >
        <div
          class="header-input row"
        >
          <div
            class="col-sm-2"
          >
            <img
              alt="search"
              class="search-icon"
              src="search.png"
            />
          </div>
          <div
            class="col-sm-6"
          >
            <div
              class="input-group"
            >
              <input
                class=" mr-sm-2 form-control"
                placeholder="Sort by:"
                type="text"
              />
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                >
                  <img
                    alt="dropdown"
                    src="dropdown.png"
                  />
                </span>
              </div>
            </div>
          </div>
          <div
            class="col"
          >
            <img
              alt="filter"
              class="filter-icon"
              src="funnel.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
